src/snac/server/database/SQL.php

```diff
@@ -4034,6 +4034,51 @@ public function selectTerm($termID)
     }
 
 
+    /**
+     * Get a single vocabulary record by Type and Value
+     *
+     * @param string $value The value of a vocabulary term
+     *
+     * @param string $type The type of a vocabulary term
+     *
+     * @return string[] A list with keys: id, type, value, uri, description
+     *
+     */
+    public function selectTermByValueAndType($value, $type)
+    {
+        $qq = 's_term_by_value';
+        $this->sdb->prepare($qq,
+                            'select
+                            id, type, value, uri, description
+                            from vocabulary where value=$1 and type=$2');
+        $result = $this->sdb->execute($qq, array($value, $type));
+        $row = $this->sdb->fetchrow($result);
+        $this->sdb->deallocate($qq);
+        return $row;
+    }
+
+    /**
+     * Get a single vocabulary record by URI
+     *
+     * @param string $uri The uri of a vocabulary term
+     *
+     * @return string[] A list with keys: id, type, value, uri, description
+     *
+     */
+    public function selectTermByUri($uri)
+    {
+        $qq = 's_term_by_uri';
+        $this->sdb->prepare($qq,
+                            'select
+                            id, type, value, uri, description
+                            from vocabulary where uri=$1');
+        $result = $this->sdb->execute($qq, array($uri));
+        $row = $this->sdb->fetchrow($result);
```

Comment:
Same as above.  I'm wondering now if selectTerm() had a provision for what happens if the ID isn't found.

Reply:
All three selectTerm..() functions return an empty row if the match isn't found. Right now, that's caught in DBUtil populateTerm() line 1718. What would be the best way to catch that in the SQL?

Reply:
I think I overthought.  fetchRow in DatabaseConnector just uses pg_fetch_all in PHP, which returns false if there is no row.  So, my concern of what happens when we try to get a row that doesn't exist isn't well founded.  Leaving it as is works well, sorry for the confusion.